bears/vcs/git/GitRevertInspectBear.py

```diff
@@ -0,0 +1,126 @@
+        for result in dependency_results[GitCommitMetadataBear.name]:
+
+            m = self.GIT_REVERT_COMMIT_RE.match(result.raw_commit_message)
+            if not m:
+                return
+
+            if not allow_git_revert_commit:
+                yield Result(self, 'Revert commit is not allowed.')
+                return
+
+            reverted_commit_sha = m.group(1)
+            get_files_command = ('git show --pretty="" --name-status %s' %
+                                 (reverted_commit_sha))
+
+            all_files = run_shell_command(get_files_command)[0]
+            files_modified_by_reverted_commit = all_files.split('\n')
+
+            reverted_commit_modified_files_list = []
+            reverted_commit_added_files_list = []
+            reverted_commit_deleted_files_list = []
+
+            for line in files_modified_by_reverted_commit:
+                pos = line.find('\t')
+                change = line[:pos]
+                if change == 'M':
+                    reverted_commit_modified_files_list.append(line[pos+1:])
+                elif change == 'A':
+                    reverted_commit_added_files_list.append(line[pos+1:])
+                elif change == 'D':
+                    reverted_commit_deleted_files_list.append(line[pos+1:])
+
+            for file_name in result.added_files:
+                if file_name not in reverted_commit_deleted_files_list:
+                    yield Result(self, 'Revert commit has an added '
+                                       'file %s that is not in reverted '
+                                       'commit.' % (file_name))
+
+            for file_name in result.deleted_files:
+                if file_name not in reverted_commit_added_files_list:
+                    yield Result(self, 'Revert commit has a deleted '
+                                       'file %s that is not in reverted '
+                                       'commit.' % (file_name))
+
+            for file_name in result.modified_files:
+                if file_name in reverted_commit_modified_files_list:
+                    head_commit_branch = run_shell_command(
+                        'git rev-parse --abbrev-ref HEAD')[0].strip('\n')
+
+                    get_revert_commit_diff_command = ('git diff HEAD^ HEAD ' +
+                                                      file_name)
+
+                    revert_commit_diff = run_shell_command(
+                        get_revert_commit_diff_command)[0]
+
+                    create_new_branch_command = ('git checkout -b ' +
+                                                 'inspectrevertbranch')
+                    run_shell_command(create_new_branch_command)
+
+                    create_expected_revert_commit = ('git revert %s --no-'
+                                                     'edit' %
+                                                     reverted_commit_sha)
+                    expected_revert_commit, err = run_shell_command(
```

Comment:
This is not a diff, this would contain something like
```
[inspectrevertbranch d214135] Revert "modify testfile4"
 Date: Tue Aug 7 20:04:17 2018 +0700
 1 file changed, 1 deletion(-)
```